packages/geo/src/lib/query/shared/query.service.ts

```diff
@@ -617,4 +605,31 @@ export class QueryService {
 
     return mime;
   }
+
+  getAllowedFieldsAndAlias(layer: any){
+    let allowedFieldsAndAlias;
+    if (
+      layer.options &&
+      layer.options.sourceOptions &&
+      layer.options.sourceOptions.sourceFields &&
+      layer.options.sourceOptions.sourceFields.length >= 1
+    ) {
+      allowedFieldsAndAlias = {};
+      layer.options.sourceOptions.sourceFields.forEach(sourceField => {
+        const alias = sourceField.alias ? sourceField.alias : sourceField.name;
+        allowedFieldsAndAlias[sourceField.name] = alias;
+      });
+    } else if (layer.options &&
+      layer.options.source &&
+      layer.options.source.options &&
+      layer.options.source.options.sourceFields &&
+      layer.options.source.options.sourceFields.length >= 1){
+        allowedFieldsAndAlias = {};
+        layer.options.source.options.sourceFields.forEach(sourceField => {
+          const alias = sourceField.alias ? sourceField.alias : sourceField.name;
+          allowedFieldsAndAlias[sourceField.name] = alias;
+        });
+      }
```

Comment:
to be replace.  
i was searching where "sourceOptions" is set ? and why not to reference to "source.options" ?   
vector layer to not have this properties and "sourceOptions" seems to be a duplicate of "source.options"
@mbarbeau